packages/drivers/odsp-driver/src/snapshotFormatConverter.ts

```diff
@@ -0,0 +1,279 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert, stringToBuffer } from "@fluidframework/common-utils";
+import { IBlob, ISnapshotTree } from "@fluidframework/protocol-definitions";
+import { ISequencedDeltaOpMessage } from "./contracts";
+import { BlobCore, NodeCore, ReadBuffer, TreeBuilder } from "./snapshotRepresentation";
+
+const minReadVersion = "1.0";
+
+/**
+ *  Header of the downloaded snapshot.
+ */
+export interface ISnapshotHeader {
+    // This is the minimum version of the reader required to read the wire format of a given snapshot.
+    minReadVersion: string,
+    // Represents the version with which snapshot is created.
+    createVersion: string,
```

Comment:
That's probably more a question to @sunnyhuang2008 & William:
I'm not sure I follow precise meaning of these two version fields (the reference to reader in particular)
I believe what we are trying to do here is mention format version used, using semantic versioning equivalent - major & minor numbers.

Reply:
Correct, the minReadVersion is the format version that the client can safely read without breaking. For example, if we were to add some features to the format, the write version would be lets say v2, but because there is no breaking change, we will specify the min read version to be the same (lets say v1) and the client will be able to still read with a v1 reader without breaking but just won't get the features of v2 without upgrading the reader. It was inspired by the "version made by" and "Version needed to Extract" from the Zip format: https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

Reply:
It would be great to follow up here and see if <major>.<minor> version solves the problem, as it's standard way to deal with these problems.
Zip file format is rather old, and I think the world moved on to leveraging semantic versioning.

Reply:
I see, with semantic versioning we can resolve the problem as well, with major versions as breaking changes, minor and patch versions as non breaking. The client would have to just be on the same major version to be able to read properly. In this case, the property bag will have string "version" that contains the semantic version of the format instead of the two minRead and CreateVersions. + @William Tao and @Marcelo Mas to comment if there are any other concerns with semantic versions.